homeassistant/components/iotty/switch.py

```diff
@@ -0,0 +1,181 @@
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IottyConfigEntry
+from .api import IottyProxy
+from .const import DOMAIN
+from .coordinator import IottyDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: IottyConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Activate the iotty LightSwitch component."""
+    _LOGGER.debug("Setup SWITCH entry id is %s", config_entry.entry_id)
+
+    coordinator: IottyDataUpdateCoordinator = config_entry.runtime_data.coordinator
+    entities = [
+        IottyLightSwitch(
+            coordinator=coordinator, iotty_cloud=coordinator.iotty, iotty_device=d
+        )
+        for d in coordinator.data.devices
+        if d.device_type == LS_DEVICE_TYPE_UID
+        if (isinstance(d, LightSwitch))
+    ]
+    _LOGGER.debug("Found %d LightSwitches", len(entities))
+
+    async_add_entities(entities)
+
+    known_devices: set = config_entry.runtime_data.known_devices
+    for known_device in coordinator.data.devices:
+        if known_device.device_type == LS_DEVICE_TYPE_UID:
+            known_devices.add(known_device)
+
+    @callback
+    def async_update_data() -> None:
+        """Handle updated data from the API endpoint."""
+        if not coordinator.last_update_success:
+            return None
+
+        devices = coordinator.data.devices
+        entities = []
+        known_devices: set = config_entry.runtime_data.known_devices
+
+        # Add entities for devices which we've not yet seen
+        for device in devices:
+            if (
+                any(d.device_id == device.device_id for d in known_devices)
+                or device.device_type != LS_DEVICE_TYPE_UID
+            ):
+                continue
+
+            iotty_entity = IottyLightSwitch(
+                coordinator=coordinator,
+                iotty_cloud=coordinator.iotty,
+                iotty_device=LightSwitch(
+                    device.device_id,
+                    device.serial_number,
+                    device.device_type,
+                    device.device_name,
+                ),
+            )
+
+            entities.extend([iotty_entity])
+            known_devices.add(device)
+
+        async_add_entities(entities)
+
+    # Add a subscriber to the coordinator to discover new devices
+    coordinator.async_add_listener(async_update_data)
+
+
+class IottyLightSwitch(SwitchEntity, CoordinatorEntity[IottyDataUpdateCoordinator]):
+    """Haas entity class for iotty LightSwitch."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_entity_category = EntityCategory.CONFIG
```

Comment:
I don't think this is correct, it should be `None`?

Documentation here: https://developers.home-assistant.io/docs/core/entity?_highlight=entitycategory#registry-properties

Reply:
Yes, having a more thorough look at the documentation and after a couple of local tests I agree with you. It has been changed to `None`